root/static/scripts/common/components/CommonsImage.js

```diff
@@ -12,48 +12,55 @@ import * as React from 'react';
 import {minimalEntity} from '../../../../utility/hydrate.js';
 import entityHref from '../utility/entityHref.js';
 
-type Props = {
-  +cachedImage: ?CommonsImageT,
-  +entity: NonUrlRelatableEntityT,
-};
-
-type State = {
-  image: ?CommonsImageT,
-};
-
-class CommonsImage extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {image: props.cachedImage};
-  }
-
-  componentDidMount() {
-    if (!this.state.image) {
-      const $ = require('jquery');
-      $.get(entityHref(this.props.entity, '/commons-image'), data => {
-        this.setState({image: data.image});
-      });
+type CommonsImageRequestCallbackT = (CommonsImageT | null) => void;
+
+function loadCommonsImage(
+  entity: NonUrlRelatableEntityT,
+  callback: CommonsImageRequestCallbackT,
+): void {
+  const url = entityHref(entity, '/commons-image');
+
+  fetch(url)
+    .then(resp => resp.json())
+    .then((reqData) => {
+      callback(reqData.image);
+    })
+    .finally(() => {
+    });
+}
+
+component CommonsImage(
+  cachedImage: ?CommonsImageT,
+  entity: NonUrlRelatableEntityT,
+) {
+  const [commonsImage, setCommonsImage] = React.useState(cachedImage);
+
+  const loadCallback: CommonsImageRequestCallbackT =
+  React.useCallback((data) => {
+    setCommonsImage(data);
+  }, [setCommonsImage]);
```

Comment:
Should be indented since it's a continuation of the previous line (I'm surprised our 20,000 ESLint rules allow this?)
```suggestion
    React.useCallback((data) => {
      setCommonsImage(data);
    }, [setCommonsImage]);
```